core/client/src/main/java/alluxio/client/block/StreamFactory.java

```diff
@@ -43,6 +44,10 @@ private StreamFactory() {} // prevent instantiation
    */
   public static BlockOutStream createLocalBlockOutStream(FileSystemContext context, long blockId,
       long blockSize, WorkerNetAddress address, OutStreamOptions options) {
+    if (NettyClient.isDomainSocketSupported(address)) {
+      return BlockOutStream
+          .createNettyBlockOutStream(blockId, blockSize, address, context, options);
+    }
```

Comment:
I think `Local` no longer makes sense since the Netty stream with domain socket could also be reading from a local machine.